xarray/tests/test_backends.py

```diff
@@ -563,6 +563,35 @@ def test_encoding_same_dtype(self):
             self.assertEqual(actual.x.encoding['dtype'], 'f4')
         self.assertEqual(ds.x.encoding, {})
 
+    @contextlib.contextmanager
+    def roundtrip_append(self, data, save_kwargs={}, open_kwargs={},
+                         allow_cleanup_failure=False):
+        with create_tmp_file(
+                allow_cleanup_failure=allow_cleanup_failure) as tmp_file:
+            for i, key in enumerate(data.variables):
+                mode = 'a' if i > 0 else 'w'
+                data[[key]].to_netcdf(tmp_file, mode=mode, **save_kwargs)
+            with open_dataset(tmp_file,
+                              autoclose=self.autoclose, **open_kwargs) as ds:
```

Comment:
@shoyer - I've put these tests in a base class now. I've never quite understood how these tests work though. Are the inherited classes passing open/save_kwargs that specify the engine? How does this work?

Reply:
Yes, that's exactly right.

One day it would be nice to split apart the gigantic `test_backends.py` into separate files for each backend...